doc/rst/developer/bestPractices/buildingdocs.rst

```diff
@@ -1,15 +1,227 @@
+============================================
+
+The file chapel/doc/rst/index.rst is the root file that sets up the
+structure of the documentation and results in the current sidebar
+displayed on https://chapel-lang.org/docs/.
+
+
+There is a strong correlation in side bar of
+https://chapel-lang.org/docs/ and the doc/rst/ subdirectories,
+except for lots of the stuff under Writing Chapel Programs.
+
+Below, the various kinds of documentation are organized by how they are
+converted into .html files.  The first column indicates the document name
+in the sidebar of https://chapel-lang.org/docs/, the second column shows
+the source files in the repository, and the last column indicates the sub
+heading the file is under.
+
+.. list-table:: **Using Sphinx to convert .rst files to .html files**
+   :widths: 100 100 100
+   :header-rows: 1
+
+   * - Doc name in Sidebar
+     - Source file(s)
+     - Sidebar subheading
+   * - Quickstart Instructions
+     - doc/rst/usingchapel/QUICKSTART.rst
+     - COMPILING AND RUNNING CHAPEL
+   * - Using Chapel
+     - doc/rst/usingchapel/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Platform-Specific Notes
+     - doc/rst/platforms/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Technical Notes
+     - doc/rst/technotes/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Tools
+     - doc/rst/tools/*
+     - COMPILING AND RUNNING CHAPEL
+   * - Quick Reference
+     - doc/rst/language/reference.rst
+     - WRITING CHAPEL PROGRAMS
+   * - Mason Packages
+     - doc/rst/mason-packages.rst
+     - WRITING CHAPEL PROGRAMS
+   * - Chapel Users Guide (WIP)
+     - doc/rst/users-guide
+     - WRITING CHAPEL PROGRAMS
+   * - Chapel Evolution
+     - doc/rst/language/evolution.rst
+     - LANGUAGE HISTORY
+   * - Documentation Archives
+     - doc/rst/language/archivedSpecs.rst
+     - LANGUAGE HISTORY
+
+
+.. list-table:: **Use chpl2rst.py on .chpl files and then sphinx to generate html**
+   :widths: 100 100 100
+   :header-rows: 1
+
+   * - Doc name in Sidebar
+     - Source file(s)
+     - Sidebar subheading
+   * - Hello World Variants
+     - doc/rst/meta/examples/index.rst + test/release/examples/hello*.chpl
+     - WRITING CHAPEL PROGRAMS
+   * - Primers
+     - doc/rst/meta/primers/index.rst + test/release/examples/primers/\*.chpl
+     - WRITING CHAPEL PROGRAMS
+
+
+
+.. list-table:: **Use chpldoc on .chpl files and then sphinx to generate html**
+   :widths: 100 100 100
+   :header-rows: 1
+
+   * - Built-in Types and Functions
+     - doc/rst/builtins + modules/internal/\*.chpl
+     - WRITING CHAPEL PROGRAMS
+   * - Standard Modules
+     - doc/rst/modules/standard + modules/standard/\*.chpl
+     - WRITING CHAPEL PROGRAMS
+   * - Package Modules
+     - doc/rst/modules/packages + modules/packages/\*.chpl
+     - WRITING CHAPEL PROGRAMS
+   * - Standard Layouts and Distributions
+     - doc/rst/modules/layoutdist + modules/layouts/\*.chpl + modules/dists/\*.chpl
+     - WRITING CHAPEL PROGRAMS
```

Comment:
I wonder whether this would be easier to sort through if the order of the items reflected the order in the sidebar and then an additional column indicated one of the three techniques for building the docs?  Though maybe it's not worth the trouble (I ask because at first I was puzzled about why a bunch of entries were missing from the first table).  A search should get people where they need to be.